File: dailyQuestion/2020-09/9-18/golang/solution_dp.go
package main

import (
	"fmt"
)

func main() {
	result := countNumbersWithUniqueDigits(2)
	fmt.Println(result)
}

func countNumbersWithUniqueDigits(n int) int {
	if n == 0 {
		return 1
	}
	first, second := 10, 9
	for i := 1; i < n; i++ {
		second *= 10 - i
		first += second
	}
	return first
}

File: dailyQuestion/2020-09/9-18/python/solution_dp.py
def countNumbersWithUniqueDigits(n: int) -> int:
    if n == 0 :
        return 1
    first, second = 10, 9
    for i in range(1,n) :
        second *= 10-i
        first += second
    return first

if __name__ == "__main__" :
    result = countNumbersWithUniqueDigits(3)
    print(result)
